### PR TITLE
Fix App Engine deployment docs

### LINKED ISSUE #8035: Incorrect instructions for deploying to Google App Engine flexible environment
**Describe the bug**
The instructions at https://strapi.io/documentation/3.0.0-beta.x/deployment/google-app-engine.html work for App Engine standard environment, but doesn't work for flexible environment.

**Steps to reproduce the behavior**

1. Follow guide
2. See error when executing `gcloud app deploy app.yaml --project myapi-123456`

**Expected behavior**
The deploy is successful.

**System**
_Not relevant_

- Node.js version: 
- NPM version:
- Strapi version: 
- Database:
- Operating system:

**Additional context**
The app.yaml for flexible environment has incorrect runtime, it should be `nodejs` instead of `nodejs10`, the latter leads to deployment failure with error message `ERROR: (gcloud.app.deploy) Unable to resolve a builder for runtime: [nodejs10]`

Also adding `app.yaml` .gcloudignore will cause the build to fail with the following error: `Application detection failed: Error: The file app.yaml does not exist`

### PR DESCRIPTION
**Pull request by akerekes**
FIXES #8035 

### What does it do?

Updates docs for App Engine Flexible deployments:
 - Fix the runtime string for App Engine Flexible.
 - Remove instructions to add 'app.yaml` to `.gcloudignore`

### Why is it needed?

The current docs show incorrect runtime setting and add `app.yaml` to `.gcloudignore` which can cause build failure.

### Related issue(s)/PR(s)

N/A

### CODE REVIEW

File: docs/v3.x/deployment/google-app-engine.md [hunk 1]
```diff
@@ -93,8 +93,6 @@ Create another database, named `strapi` for example. It may be useful to delete
 
 Create the `app.yaml` file in the project root.
 
-Add `app.yaml` to `.gitignore`.
```
Comment (MattieBelt) [inline docs/v3.x/deployment/google-app-engine.md:96]: One might argue that this way database secrets could end up in the repo and be leaked.

Comment (akerekes) [reply docs/v3.x/deployment/google-app-engine.md:96]: That's correct, this was removed in error.

**Review by derrickmehaffy — APPROVED** [review]
LGTM, can you sign the CLA (we just added it)

**Comment by akerekes** [comment]
> LGTM, can you sign the CLA (we just added it)

Signed.

**Review by derrickmehaffy — APPROVED** [review]
LGTM

**Comment by akerekes** [comment]
What else is needed to merge this PR?

**Comment by derrickmehaffy** [comment]
Closing as we moved to new doc repo, manually synced PR over here: https://github.com/strapi/documentation/pull/24

Thank you @akerekes

**Pull request closed without merging**